src/vivarium_nih_us_cvd/components/risks.py

```diff
@@ -95,3 +126,55 @@ def _get_current_exposure(self, index: pd.Index) -> pd.Series:
         return (
             self.gbd_exposure(index) * self.population_view.get(index)[COLUMNS.SBP_MULTIPLIER]
         )
+
+    def _get_target_modifier(
+        self, builder: Builder
+    ) -> Callable[[pd.Index, pd.Series], pd.Series]:
+        """Apply medication effects"""
+
+        def adjust_target(index: pd.Index, target: pd.Series) -> pd.Series:
+            """Determine the exposure decrease as treatment_efficacy * adherence_score"""
+            pop_view = self.population_view.get(index)
+            adherence_scores = pop_view[COLUMNS.SBP_MEDICATION_ADHERENCE].map(
+                MEDICATION_ADHERENCE_SCORE
+            )
+            df_efficacy = pd.DataFrame(
+                {"bin": pd.cut(x=target, bins=sbp_bin_edges, right=True)}
+            )
+            df_efficacy["sbp_start_exclusive"] = df_efficacy["bin"].apply(lambda x: x.left)
+            df_efficacy["sbp_end_inclusive"] = df_efficacy["bin"].apply(lambda x: x.right)
+            df_efficacy = pd.concat([df_efficacy, pop_view[COLUMNS.SBP_MEDICATION]], axis=1)
+            df_efficacy = (
+                df_efficacy.reset_index()
+                .merge(
+                    sbp_risk_effects,
+                    on=["sbp_start_exclusive", "sbp_end_inclusive", COLUMNS.SBP_MEDICATION],
+                    how="left",
+                )
+                .set_index("index")
+            )
+            # Simulants not on treatment mean 0 effect
+            assert set(
+                df_efficacy.loc[df_efficacy["value"].isna(), COLUMNS.SBP_MEDICATION]
+            ) == {SBP_MEDICATION_LEVEL.NO_TREATMENT.DESCRIPTION}
+            df_efficacy.loc[
+                df_efficacy[COLUMNS.SBP_MEDICATION]
+                == SBP_MEDICATION_LEVEL.NO_TREATMENT.DESCRIPTION,
+                "value",
+            ] = 0
+            assert df_efficacy["value"].isna().sum() == 0
```

Comment:
Would these asserts ever be raised?

Reply:
Not sure I understand the question. If the assertion fails it had better break b/c something went wrong

Reply:
Is this just a check for sanity sake?  The assertion will break if something goes wrong but shouldn't you handle this when you are processing before you hit the assert statement?    This is a good check but it will just break and you will have to go back and find the source of the issue if you are expected these actually be hit.

Reply:
In other words, disregard this comment if you aren't expecting to to raise the assert statement but just added them to be safe.